x/ante/check_refund.go

```diff
@@ -56,28 +59,32 @@ func (d CheckRefundFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 			feePayer = feeTx.FeePayer()
 		}
 
-		req := msgs[0].(*rewardtypes.RefundMsgRequest)
-		err := d.reward.SetPendingRefund(ctx, *req, rewardtypes.Refund{Payer: feePayer, Fees: fees})
-		if err != nil {
-			return ctx, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, err.Error())
+		splitFees := d.splitFeesEvenly(refundableCount, fees)
+
+		for _, msg := range msgs {
+			switch msg := msg.(type) {
+			case *rewardtypes.RefundMsgRequest:
+				req := *msg
+				err := d.reward.SetPendingRefund(ctx, req, rewardtypes.Refund{Payer: feePayer, Fees: splitFees[0]})
```

Comment:
batched refund msgs could cause dangling PendingRefunds. 

antehandler set refunds, pending refunds won't be cleared if the reward message execution failed.